compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1281,19 +1297,21 @@ convertArg env = \case
     Type t -> TyArg <$> convertType env t
     e -> TmArg <$> convertExpr env e
 
-withTyArg :: Env -> (LF.TypeVarName, LF.Kind) -> [LArg Var] -> (LF.Type -> [LArg Var] -> ConvertM (LF.Expr, [LArg Var])) -> ConvertM (LF.Expr, [LArg Var])
+withTyArg :: Env -> LF.Kind -> [LArg Var] -> (Env -> LF.Type -> [LArg Var] -> ConvertM (LF.Expr, [LArg Var])) -> ConvertM (LF.Expr, [LArg Var])
 withTyArg env _ (LType t:args) cont = do
     t <- convertType env t
-    cont t args
-withTyArg env (v, k) args cont = do
-    (x, args) <- cont (TVar v) args
+    cont env t args
+withTyArg env k args cont = do
+    let (v, env') = envFreshTypeVar "a" env
```

Comment:
Should we still use something like `$$t` here, as the name hint? It probably doesn't make a difference, given how careful we are with type variables, but it has less chance of colliding with other type variables already in the code.

Reply:
Yeah, that's a good point. I'll use `$t` to save some money.